src/NServiceBus.Core/Routing/RoutingSettings.cs

```diff
@@ -38,16 +38,8 @@ public void RouteToEndpoint(Type messageType, string destination)
         public void RouteToEndpoint(Assembly assembly, string destination)
         {
             ThrowOnAddress(destination);
-            Settings.GetOrCreate<ConfiguredUnicastRoutes>().Add((routingTable, knownMessageTypes) =>
-            {
-                foreach (var knownMessage in knownMessageTypes)
-                {
-                    if (knownMessage.Assembly == assembly)
-                    {
-                        routingTable.RouteTo(knownMessage, UnicastRoute.CreateFromEndpointName(destination));
-                    }
-                }
-            });
+
+            Settings.GetOrCreate<ConfiguredUnicastRoutes>().Add(new AssemblyRouteSource(assembly, Settings.Get<Conventions>(), UnicastRoute.CreateFromEndpointName(destination)));
```

Comment:
we need to pass the conventions as a parameter during feature setup because at configuration time the final conventions may not be configured yet.